src/module/vector.jl

```diff
@@ -94,13 +98,13 @@ end
 #
 ###############################################################################
 
-function *{T <: Nemo.RingElem}(a::svector{spoly{T}}, b::spoly{T})
-   check_parent(a, b)
+function *(a::svector{spoly{T}}, b::spoly{T}) where T <: AbstractAlgebra.RingElem
+   base_ring(a) != parent(b) && error("Incompatible base rings")
    R = base_ring(a)
    a1 = libSingular.p_Copy(a.ptr, R.ptr)
    b1 = libSingular.p_Copy(b.ptr, R.ptr)
    s = libSingular.p_Mult_q(a1, b1, R.ptr)
-   return Singular.Vector(R, a.rank, s)
+   return svector{spoly{T}}(R, a.rank, s)
 end
 
 *{T <: Nemo.RingElem}(a::spoly{T}, b::svector{spoly{T}}) = b*a
```

Comment:
Here you assume, that the module is two-sided.
How do we handle the non-commutative case (currently not in Singular.jl,
but Singular provides that) ?